test-sol/strategies/aaveV3/TestAaveV3Automation.t.sol

```diff
@@ -0,0 +1,352 @@
+
+        AaveV3SubProxy.AaveSubData memory params = user.subToAutomationBundles(
+            _isSafe,
+            address(subProxy),
+            MIN_RATIO,
+            MAX_RATIO,
+            TARGET_RATIO_BOOST,
+            TARGET_RATIO_REPAY
+        );
+
+        repaySubId = SubStorage(SUB_STORAGE_ADDR).getSubsCount() - 2;
+        boostSubId = SubStorage(SUB_STORAGE_ADDR).getSubsCount() - 1;
+
+        repaySub = _formatRepaySub(params);
+        boostSub = _formatBoostSub(params);
+    }
+
+    function _walletSetUpBeforeEachTest(bool _isSafe) internal {
+        wallet = _isSafe ? user.safeAddr() : user.proxyAddr();
+        _createAaveV3Position(_isSafe, wallet);
+        _subToAutomationBundles(_isSafe);
+    }
+
+    //////////////////////////////// TESTs /////////////////////////////////////////////
+
+    function testAaveV3RepayStrategy() public {
+        _testAaveV3RepayStrategy(false);
+        _testAaveV3RepayStrategy(true);
+    }
+
+    function testAaveV3FLRepayStrategy() public {
+        _testAaveV3FLRepayStrategy(false);
+        _testAaveV3FLRepayStrategy(true);
+    }
+
+    function testAaveV3BoostStrategy() public {
+        _testAaveV3BoostStrategy(false);
+        _testAaveV3BoostStrategy(true);
+    }
+
+    function testAaveV3BoostFLStrategy() public {
+        _testAaveV3BoostFLStrategy(false);
+        _testAaveV3BoostFLStrategy(true);
+    }
+
+    function _testAaveV3RepayStrategy(bool _isSafe) internal {
+        _walletSetUpBeforeEachTest(_isSafe);
+
+        uint beforeRatio = trigger.getSafetyRatio(AAVE_MARKET, wallet);
+
+        (, uint256 borrowAmountBefore,,,, ) = pool.getUserAccountData(wallet);
+        uint256 txFeeBalanceBefore = IERC20(TokenAddresses.WETH_ADDR).balanceOf(TokenAddresses.FEE_RECEIVER);
+
+        bytes[] memory _triggerCallData = new bytes[](1);
+
+        bytes[] memory _actionsCallData = new bytes[](5);
+        _actionsCallData[0] = aaveV3WithdrawEncode(collateralAsset.id, true, REPAY_AMOUNT_WETH, wallet, address(0));
+        _actionsCallData[1] = sellEncode(TokenAddresses.WETH_ADDR, TokenAddresses.DAI_ADDR, 0, wallet, wallet, TokenAddresses.UNI_V2_WRAPPER);
+        _actionsCallData[2] = gasFeeEncode(REPAY_GAS_COST, TokenAddresses.DAI_ADDR);
+        _actionsCallData[3] = aaveV3PaybackEncode(0, wallet, 2, debtAsset.id,true, false, address(0), address(0));
+        _actionsCallData[4] = aaveV3RatioCheckEncode(0, 0);
+
+        executor.executeStrategy(repaySubId, INDEX_REPAY, _triggerCallData, _actionsCallData, repaySub);
+
+        uint afterRatio = trigger.getSafetyRatio(AAVE_MARKET, wallet);
+
+        uint256 txFeeBalanceAfter = IERC20(TokenAddresses.WETH_ADDR).balanceOf(TokenAddresses.FEE_RECEIVER);
+        (, uint256 borrowAmountAfter,,,, ) = pool.getUserAccountData(wallet);
+
+        uint amountAfterFee = REPAY_AMOUNT_WETH - (REPAY_AMOUNT_WETH / 400);
+
+        assertEq(REPAY_AMOUNT_WETH - amountAfterFee, txFeeBalanceAfter - txFeeBalanceBefore);
```

Comment:
Love the fact you added an additional assertion here, rather than just relying on the standard before-and-after ratio check : )